src/lib.rs

```diff
@@ -229,9 +251,48 @@ pub fn report_tracemap(configs: &[Config], tracemap: TraceMap) -> Result<(), Run
     Ok(())
 }
 
-fn report_coverage_with_check(c: &Config, tracemap: &TraceMap) -> Result<(), RunError> {
-    report_coverage(c, tracemap)?;
-    check_fail_threshold(tracemap, c)
+fn check_coverage(delta: f64, config: &Config, trace: &TraceMap) -> Result<(), RunError> {
+    let threshold = check_fail_threshold(trace, config);
```

Comment:
You can make this function a lot simpler by doing just:

```rust
	check_fail_threshold(trace, config)?;
        if config.fail_under.is_none() {
	    check_fail_decreasing(delta, config)?;
        }
         Ok(())
```

I think this covers all the feature space, you'd just have to move the error log into `check_fail_decreasing`

Reply:
I'm not sure that would be correct with the current feature set ?
If I understand the change request correctly, then check_coverage would return the threshold error, regardles if the coverage is increasing, could you read the commit message on the last commit with the fail-under integration and see if those cases are still coveret, which I don't think they are.

Have forgotten to update the pull request descrition

Reply:
Well if you set `--fail-under` then I would expect it to always fail if below that coverage amount even if the coverage is increasing because you haven't met the threshold.

Reply:
My thought was, that when you start to use tarpaulin, that you would be able to use it at once, and you set your end target from the start and don't have to correct it later, by providing kind of a "soft start", so that you don't have to provide all test cases at once, but you will have to provide tests for all new code.
Once you reach the threshold, then --fail-decreasing is basically not used anymore with the code implemented, if used in combination with --fail-under, where it takes over.
The user is always welcome to use them individually if the combined functionality is not what they want.

Reply:
Hmm I think that just adds confusion to the flag. If you don't want to see the behaviour at the start of a project you can just omit it. Though come to think of it most users use tarpaulin in CI and this flag won't do anything then anyway as they typically won't cache the previous tarpaulin results.